### PR TITLE
Jackson by default. Remove ScalaJS support.

### PR DESCRIPTION
**Pull request by htmldoug**
## Summary
- Replaces all use of the pure scala json parser/generator with Jackson.
- scala parser is sidelined to a new orphaned jar called "weejson-parser" and is not published.
- Removes ScalaJS support.
- Avoids changing class names/methods as much as possible. Mostly just cuts out the scala parser and replaces in place to minimize changes. API changes should be a separate PR.

## Jackson
There are places that the jackson parser is [5x faster](https://github.com/rallyhealth/weePickle/blob/b088175f380bd47e830376b909e064052c85cdfc/bench/src-jvm/bench/JmhBench.scala#L24-L25) than the pure scala versions. Migrating to this library should NOT cause perf regressions.

I've tested jackson-databind 2.7 (and friends) with jackson-core 2.10 and encountered no issues. My assumption is that the jackson-core API is "stable enough".

### ScalaJS Removal
As a java library, jackson-core does not run on scalajs. I briefly explored transpiling it to scala, but https://github.com/timowest/scalagen choked on java generics (e.g. `ArrayList<String>`). The IntelliJ converter did much better, but can't translate java enums to scala.

I also explored refactoring weepickle to have `src-jvm` use jackson and `src-js` use the scala parser. This is possible, but the refactoring required is larger than I have time for. Such refactoring would be a goal for v2.

Rally Health doesn't currently use ScalaJS and isn't likely to for the foreseeable future.

### Project structure
After deep diving on project structure, I want to capture my thoughts for later.

http://www.lihaoyi.com/upickle/#CustomConfiguration "Api bundle" is novel, and very clever, but complicates other things, and likely YAGNI. For example, `SnakePickle` in the docs could be accomplished by a `SnakeVisitor`.

By contrast, play-json defines `trait Reads[T]`, `trait Writes[T]` lets you put type classes instances in your companion objects. I'd like to follow this model with the `Value`, `IndexedValue`, and `Msg` ASTs, but I can't because implementing `Reader[T]` and `Writer[T]` in `weejson.jar` and `weepack.jar` would create a circular dependency on `weepickle.jar` which already depends on the other two. I'd like for this to be reversed and lean on implicit resolution for composition, effectively simplifying this to "play-json with visitors".

One complication is `Array[Byte]`. `Writer[Array[Byte]]` is ambiguous by type info alone. It could parse as json OR msg pack to emit to a visitor or just call `visitor.visitBinary(bytes)` directly. This needs further thought.

## Pluggable parsers
I'd like to decouple the json parsing behind something like play-json's [JsonFacade](https://github.com/playframework/play-json/blob/48cf1ae4744516fbe8527a42960e0804bdc0bf71/play-json/shared/src/main/scala/play/api/libs/json/Json.scala#L9-L163). Jawn is under active development again and preparing for a v1.0.0 release and may have perf improvements worth evaluating. Out of scope of this PR though.

### CODE REVIEW

File: bench/src-jvm/bench/JmhBench.scala [hunk 1]
```diff
@@ -50,22 +49,19 @@ class JmhBench {
...
   }
 
-  @Benchmark
-  def bytesToCcWeeJson(bh: Blackhole): Unit = {
-    bh.consume(WeePickle.read[Seq[Common.Data]](Common.benchmarkSampleJsonBytes))
-  }
+  //  @Benchmark
+  //  def bytesToCcWeeJson(bh: Blackhole): Unit = {
+  //    bh.consume(WeePickle.read[Seq[Common.Data]](Common.benchmarkSampleJsonBytes))
+  // TODO needs reimplementing
```
Comment (htmldoug) [inline bench/src-jvm/bench/JmhBench.scala]: - [ ] Should be hooked back up in this PR.

File: weepickle/test/src/com/rallyhealth/weepickle/v1/example/ExampleTests.scala [hunk 2]
```diff
@@ -394,27 +393,29 @@ object ExampleTests extends TestSuite {
...
-      WeePack.transform(msg, new com.rallyhealth.weejson.v0.StringRenderer()).toString ==>
-        """[{"myFieldA":1,"myFieldB":"g"},{"myFieldA":2,"myFieldB":"k"}]"""
-
-      WeePack.transform(binary, new com.rallyhealth.weejson.v0.StringRenderer()).toString ==>
-        """[{"myFieldA":1,"myFieldB":"g"},{"myFieldA":2,"myFieldB":"k"}]"""
-
-      // Some messagepack structs cannot be converted to valid JSON, e.g.
-      // they may have maps with non-string keys. These can still be pretty-printed:
-      val msg2 = com.rallyhealth.weepack.v0.Obj(com.rallyhealth.weepack.v0.Arr(com.rallyhealth.weepack.v0.Int32(1), com.rallyhealth.weepack.v0.Int32(2)) -> com.rallyhealth.weepack.v0.Int32(1))
-      WeePack.transform(msg2, new com.rallyhealth.weejson.v0.StringRenderer()).toString ==> """{[1,2]:1}"""
```
Comment (htmldoug) [inline weepickle/test/src/com/rallyhealth/weepickle/v1/example/ExampleTests.scala]: This throws currently and I'm okay with that for now.

File: weepickle/test/src/com/rallyhealth/weepickle/v1/example/ExampleTests.scala [hunk 3]
```diff
@@ -571,7 +572,12 @@ object ExampleTests extends TestSuite {
     }
     test("byteArrays"){
       import com.rallyhealth.weepickle.v0.WeePickle._
-      write(Array[Byte](1, 2, 3, 4)) ==> "[1,2,3,4]"
+      /**
+        * JSON encoding isn't symmetric here,
+        * but base64 is more useful.
+        * e.g. https://stackoverflow.com/a/247261
+        */
+      write(Array[Byte](1, 2, 3, 4)) ==> """"AQIDBA==""""
```
Comment (htmldoug) [inline weepickle/test/src/com/rallyhealth/weepickle/v1/example/ExampleTests.scala]: Behavior change. Sacrificing purity for usefulness.

File: bench/src-jvm/bench/JmhBench.scala (hunk 1 above)
Comment (htmldoug) [reply bench/src-jvm/bench/JmhBench.scala]: Spot check shows results are still the same.

File: weejson/jackson/src/com/rallyhealth/weejson/v0/jackson/WeeJackson.scala [hunk 4]
```diff
@@ -17,91 +18,66 @@ import scala.util.Try
...
-  /**
-    * Parses a single JSON value using the default Jackson parser.
-    *
-    * @example
-    * {{{
-    *   WeeJackson.parse("""{"pony" -> "twilight sparkle"}""").transform(Value)
-    * }}}
-    */
-  def parse(s: String): Readable = parse(Instance.createParser(s))
+  def toGenerator(generator: JsonGenerator): Visitor[_, JsonGenerator] = new JsonGeneratorVisitor(generator)
```
Comment (russellremple) [inline weejson/jackson/src/com/rallyhealth/weejson/v0/jackson/WeeJackson.scala]: seems like a weird name

Comment (htmldoug) [reply weejson/jackson/src/com/rallyhealth/weejson/v0/jackson/WeeJackson.scala]: Yeah, the naming's a mess.

**Pull request merged**